app/main/views/organizations.py

```diff
@@ -88,6 +95,96 @@ def get_organization_message_allowance(org_id):
     }
 
 
+def _handle_create_service(org_id):
+    create_service_form = CreateServiceForm(
+        organization_type=current_user.default_organization_type
+        or OrganizationType.FEDERAL
+    )
+
+    if request.method == "POST" and create_service_form.validate_on_submit():
+        service_name = create_service_form.name.data
+        service_id, error = _create_service(
+            service_name,
+            create_service_form.organization_type.data,
+            email_safe(service_name),
+            create_service_form,
+        )
+        if not error:
+            current_organization.associate_service(service_id)
+            flash(f"Service '{service_name}' has been created", "default_with_tick")
+            session["new_service_id"] = service_id
+            return redirect(url_for(".organization_dashboard", org_id=org_id))
+        else:
+            flash("Error creating service", "error")
+
+    return create_service_form
+
+
+def _handle_invite_user(org_id):
+    invite_user_form = InviteOrgUserForm(
+        inviter_email_address=current_user.email_address
+    )
+
+    if request.method == "POST" and invite_user_form.validate_on_submit():
+        try:
+            invited_org_user = InvitedOrgUser.create(
+                current_user.id, org_id, invite_user_form.email_address.data
+            )
+            flash(
+                f"Invite sent to {invited_org_user.email_address}",
+                "default_with_tick",
+            )
+            return redirect(url_for(".organization_dashboard", org_id=org_id))
+        except Exception:
+            flash("Error sending invitation", "error")
+
+    return invite_user_form
+
+
+def _handle_edit_service(org_id, service_id):
+    service = Service.from_id(service_id)
+
+    if request.method == "POST":
+        service_name = request.form.get("service_name", "").strip()
+        primary_contact = request.form.get("primary_contact", "").strip()
+        new_status = request.form.get("status")
+
+        if not service_name:
+            flash("Service name is required", "error")
+        elif primary_contact and not re.match(EMAIL_REGEX, primary_contact):
```

Comment:
With the change above to use `re.compile()`, this needs to change slightly too (again, [referring to the Python regex documentation](https://docs.python.org/3/library/re.html#re.compile)):

```suggestion
        elif primary_contact and not EMAIL_REGEX.match(primary_contact):
```